src/core/main.c

```diff
@@ -1182,8 +1184,15 @@ static int bump_rlimit_nofile(struct rlimit *saved_rlimit) {
                 arg_default_rlimit[RLIMIT_NOFILE] = rl;
         }
 
+        /* Get current RLIMIT_NOFILE maximum compiled into the kernel. */
+        r = read_one_line_file("/proc/sys/fs/nr_open", &nr_open);
+        if (r == 0)
+                r = safe_atollu(nr_open, &min_max);
```

Comment:
ah, is that hardcoded somewhere, I wasn't aware. The 64K I placed there I actually just pulled out of my hat. If there's a hardcoded value in the kernel, it would definitely make sense to use that. Would be thankful for such a change! And please add a quick comment saying that this is a kernel hardcoded value